### PR TITLE
chore: upgrade mongodb-runner COMPASS-8181

### PR DESCRIPTION
**Pull request by kmruiz**
### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by Anemy — APPROVED** [review]

File: packages/compass/package.json [hunk 1]
```diff
@@ -181,6 +181,7 @@
   },
   "dependencies": {
     "@mongosh/node-runtime-worker-thread": "^2.3.0",
+    "@mongodb-js/mongodb-downloader": "^0.3.5",
```
Comment (gribnoysup) [inline packages/compass/package.json:184]: This will include this package with the distribution which we don't want

Comment (kmruiz) [reply packages/compass/package.json:184]: Yep, that is what I was mentioning on Slack, that his should be a devDependency, compass/scripts don't have a specific package.json.

Comment (gribnoysup) [reply packages/compass/package.json:184]: I'm not sure what you mean, <root>/scripts is a workspace in the monorepo and does have it's own package.json at <root>/scripts/package.json

Comment (kmruiz) [reply packages/compass/package.json:184]: The script that depends on the mongodb-downloader is on <root>/packages/compass/scripts/download-csfle.js, not in <root>/scripts/ and it doesn't have it's own package.json. I'm not sure why it's there, but I suppose there should be a reason.

Comment (gribnoysup) [reply packages/compass/package.json]: Ahhh, so all this time you were talking about scripts folder in compass, not scripts package, my bad. Yeah, disregard what I said about prod dependency then, this should go to the dev ones, you're correct 👍

**Review by gribnoysup — APPROVED** [review]

**Pull request merged**